server/protocols/jmap-draft/src/main/java/org/apache/james/jmap/draft/model/MailboxFactory.java

```diff
@@ -104,10 +120,29 @@ public MailboxBuilder usingPreloadedMailboxesMetadata(Optional<List<MailboxMetaD
             Preconditions.checkNotNull(session);
 
             try {
+                if (mailboxACL.isPresent() && preLoadedMailboxCounter.isPresent() && mailboxPath.isPresent()) {
```

Comment:
we usually don't build the result in branches in a build method but rather populate local variable with computed result and finally build the result.
Here, the code is very complex and it looks like a code smell to have a complex build method.
BTW, you added a lot of code path combination without adding the related tests.
It's more than likely that the tests will be painful to write because the number of combination is high. You probably will want to cut the Builder in smaller part to decrease the complexity and the number of tests.